src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -110,9 +108,8 @@ public void createTopic(final EventType eventType) throws TopicCreationException
     }
 
     private long getTopicRetentionMs(final EventType eventType) {
-        return Optional.ofNullable(eventType.getOptions())
-                .map(EventTypeOptions::getRetentionTime)
-                .orElse(settings.getDefaultTopicRetentionMs());
+        final Long retentionTime = eventType.getOptions().getRetentionTime();
+        return retentionTime == null ? settings.getDefaultTopicRetentionMs() : retentionTime.longValue();
```

Comment:
I considered the case when we want to use KafkaTopicRepository.create method and pass EventType without retention time.